paket-files/matthid/Yaaf.FSharp.Scripting/src/source/Yaaf.FSharp.Scripting/YaafFSharpScripting.fs

```diff
@@ -742,16 +731,38 @@ module internal Extensions =
           x.EvalInteraction (sprintf "open %s" ns)
       member x.Reference file =
           x.EvalInteraction (sprintf "#r @\"%s\"" file)
+      member x.Include dir =
+          x.EvalInteraction (sprintf "#I @\"%s\"" dir)
       member x.Load file =
           x.EvalInteraction (sprintf "#load @\"%s\" " file)
 
+      /// Change the current directory (so that relative paths within scripts work properly).
+      /// Returns a handle to change the current directory back to it's initial state
+      /// (Because this will change the current directory of the currently running code as well!).
+      member x.Cd dir =
+          let oldDir = System.IO.Directory.GetCurrentDirectory()
+          let cd dir =
+            x.EvalInteraction (sprintf "#cd @\"%s\"" dir)
+          cd dir
+          let isDisposed = ref false
+          { new System.IDisposable with
+              member __.Dispose() =
+                if not !isDisposed then
+                  cd oldDir
+                  isDisposed := true }
+
+      /// Same as Cd but takes a function for the scope.
+      member x.WithCd dir f =
+          use __ = x.ChangeCurrentDirectory dir
+          f ()
+          
       /// Change the current directory (so that relative paths within scripts work properly).
       /// Returns a handle to change the current directory back to it's initial state
       /// (Because this will change the current directory of the currently running code as well!).
       member x.ChangeCurrentDirectory dir =
           let oldDir = System.IO.Directory.GetCurrentDirectory()
           let cd dir =
-            x.EvalInteraction (sprintf "System.IO.Directory.SetCurrentDirectory(@\"%s\")" dir)
+            x.EvalInteraction (sprintf "System.Environment.CurrentDirectory <- @\"%s\"" dir)
```

Comment:
@matthid  Btw, I had to change this one because System.IO.Directory requires the assembly System.IO.FileSystem in the netstandard version, while System.Environment is always present.